src/crimson/osd/osd.cc

```diff
@@ -462,16 +473,20 @@ seastar::future<> OSD::stop()
   }).then([this] {
     return store->stop();
   }).then([this] {
-    return seastar::do_with(std::move(monc), std::move(mgrc),
-			    [](auto& monc, auto& mgrc) {
-      return monc->stop().then([&mgrc] {
-	return mgrc->stop();
-      });
+    auto first = pg_map.get_pgs().begin();
+    auto last = pg_map.get_pgs().end();
+    return seastar::parallel_for_each(first, last,
+      [this](auto& p) {
+      return p.second->stop();
```

Comment:
i think seastar also offer a variant which works with a range, could you use it instead?